### PR TITLE
Test plan optimization analyzer

### PR DESCRIPTION
**Pull request by devcrafter**
Update test `01655_plan_optimizations.sh` to execute with analyzer. Updated only queries which output is different with enabled analyzer

Related https://github.com/ClickHouse/ClickHouse/issues/42648

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 9ea0575ff8c7ac04d46ec93d012debf01eda55c5 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50095/9ea0575ff8c7ac04d46ec93d012debf01eda55c5/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: tests/queries/0_stateless/01655_plan_optimizations.reference [hunk 1]
```diff
@@ -129,12 +170,18 @@ Filter
 Join
 Filter column: notEquals(number, 1)
 Join
+> (analyzer) one condition of filter is pushed down before LEFT JOIN
+Join
+Filter column: notEquals(l.number_0, 1_UInt8)
```
Comment (novikd) [inline tests/queries/0_stateless/01655_plan_optimizations.reference:175]: Probably better to wait for #50097

Comment (devcrafter) [reply tests/queries/0_stateless/01655_plan_optimizations.reference:175]: @novikd should we close this PR or?

**Comment by devcrafter** [comment]
@novikd please take a look

**Review by novikd — APPROVED** [review]

**Pull request merged**